actionpack/test/dispatch/cookies_test.rb

```diff
@@ -1300,6 +1300,22 @@ def test_vanilla_cookie_with_expires_set_relatively
     end
   end
 
+  def test_purpose_metadata_for_signed_cookies
```

Comment:
We probably want the two new tests to set the `ActionDispatch::Cookies::AbstractCookieJar.use_cookies_with_metadata` config to true

Reply:
It's probably because `ActionDispatch::Cookies::AbstractCookieJar.use_cookies_with_metadata = false` was never run since the execution stoped when the assertion `assert_nil cookies.encrypted[:discount_percentage]` failed.

Reply:
I had tried both. Are tests run parallelly ?

Reply:
They are not, I tried running the tests and they work fine with something like 

<details>
  <summary> Diff </summary>

```diff
diff --git a/actionpack/test/dispatch/cookies_test.rb b/actionpack/test/dispatch/cookies_test.rb
index aa9e4dac42..9a822c4f44 100644
--- a/actionpack/test/dispatch/cookies_test.rb
+++ b/actionpack/test/dispatch/cookies_test.rb
@@ -1301,19 +1301,27 @@ def test_vanilla_cookie_with_expires_set_relatively
   end

   def test_purpose_metadata_for_signed_cookies
+    previous_config = ActionDispatch::Cookies::AbstractCookieJar.use_cookies_with_metadata
+    ActionDispatch::Cookies::AbstractCookieJar.use_cookies_with_metadata = true
     cookies.signed[:user_id] = 50
     cookies.signed[:discount_percentage] = 10

     cookies[:discount_percentage] = cookies[:user_id]
     assert_nil cookies.signed[:discount_percentage]
+  ensure
+    ActionDispatch::Cookies::AbstractCookieJar.use_cookies_with_metadata = previous_config
   end

   def test_purpose_metadata_for_encrypted_cookies
+    previous_config = ActionDispatch::Cookies::AbstractCookieJar.use_cookies_with_metadata
+    ActionDispatch::Cookies::AbstractCookieJar.use_cookies_with_metadata = true
     cookies.encrypted[:user_id] = 50
     cookies.encrypted[:discount_percentage] = 10

     cookies[:discount_percentage] = cookies[:user_id]
     assert_nil cookies.encrypted[:discount_percentage]
+  ensure
+    ActionDispatch::Cookies::AbstractCookieJar.use_cookies_with_metadata = previous_config
   end
```
</details>

Only two tests are failing related to expiry, but it makes sense since the `expiry` option is now set solely when `use_cookies_with_metadata` is `true`.

Reply:
Thanks, @Edouard-chin  for helping me out with your expertise. Got to make a few more tweaks to the PR :smile: